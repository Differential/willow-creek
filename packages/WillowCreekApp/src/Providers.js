import { nest } from 'recompose';
import { Providers } from '@apollosproject/ui-kit';
import ClientProvider from './client';

export default nest(ClientProvider, Providers);
