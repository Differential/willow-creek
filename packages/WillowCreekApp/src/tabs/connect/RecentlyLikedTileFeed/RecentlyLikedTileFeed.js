import React, { Component } from 'react';
import { View } from 'react-native';
import { withNavigation } from 'react-navigation';
import PropTypes from 'prop-types';

import {
  PaddedView,
  H4,
  UIText,
  HorizontalTileFeed,
  styled,
  ButtonLink,
  withIsLoading,
} from '@apollosproject/ui-kit';

import TileImageItem from '../../discover/TileImageItem';

const RowHeader = styled(({ theme }) => ({
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'center',
  paddingVertical: theme.sizing.baseUnit,
}))(PaddedView);

const Name = styled({
  flexGrow: 1,
})(View);

const LikedContentLink = styled({
  flexDirection: 'row-reverse',
})(View);

class RecentlyLikedTileFeed extends Component {
  static propTypes = {
    navigation: PropTypes.shape({
      navigate: PropTypes.func,
    }),
    isLoading: PropTypes.bool,
    name: PropTypes.string,
    content: PropTypes.arrayOf(
      PropTypes.any // this component doesn't care about the shape of `node`, just that it exists
    ),
  };

  loadingStateObject = {
    id: 'fake_id',
    title: '',
    coverImage: [],
  };

  titleImageItem = ({ item }) => (
    <TileImageItem
      item={item}
      isLoading={this.props.isLoading}
      navigation={this.props.navigation}
    />
  );

  render() {
    const { isLoading, name, navigation, content = [] } = this.props;

    return (
      <PaddedView horizontal={false}>
        <RowHeader>
          <Name>
            <H4 isLoading={isLoading}>{name}</H4>
          </Name>
          <LikedContentLink>
            <UIText isLoading={isLoading}>
              <ButtonLink
                onPress={() => {
                  navigation.navigate('LikedContentList');
                }}
              >
                View All
              </ButtonLink>
            </UIText>
          </LikedContentLink>
        </RowHeader>
        <HorizontalTileFeed
          initialNumToRender={5}
          content={content}
          renderItem={this.titleImageItem}
          loadingStateObject={this.loadingStateObject}
          isLoading={isLoading}
        />
      </PaddedView>
    );
  }
}

export default withNavigation(withIsLoading(RecentlyLikedTileFeed));
