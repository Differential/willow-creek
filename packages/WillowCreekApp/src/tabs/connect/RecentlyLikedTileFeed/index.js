export RecentlyLikedTileFeed from './RecentlyLikedTileFeed';
export RecentlyLikedTileFeedConnected from './RecentlyLikedTileFeedConnected';
