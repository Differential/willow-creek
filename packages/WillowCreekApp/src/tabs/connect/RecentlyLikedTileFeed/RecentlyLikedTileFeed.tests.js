import React from 'react';
import renderer from 'react-test-renderer';

import Providers from '../../../Providers';
import RecentlyLikedTileFeed from './RecentlyLikedTileFeed';

describe('RecentlyLikedTileFeed', () => {
  it('renders a RecentlyLikedTileFeed', () => {
    const props = {
      content: [
        {
          id: 'UniversalContentItem:4148aeb9482181025e9cad42826e676a',
          title: 'Making Prayer a Spiritual Habit',
          __typename: 'UniversalContentItem',
          coverImage: {
            sources: [
              {
                uri:
                  'https://apollosrock.newspring.cc/GetImage.ashx?guid=55be1fd5-d8eb-43a6-aa70-f8e27bb63d31',
              },
            ],
          },
          sharing: {
            title: 'Making Prayer a Spiritual Habit',
            message: '',
            url: 'https://apollosrock.newspring.cc/',
          },
        },
        {
          id: 'UniversalContentItem:9dbdc4e565dcaa3c9b17eb5ae42de96e',
          title: 'Vacations and Your Budget',
          __typename: 'UniversalContentItem',
          coverImage: {
            sources: [
              {
                uri:
                  'https://apollosrock.newspring.cc/GetImage.ashx?guid=993efb47-47ec-4dec-a3d3-47bcacfbd58e',
              },
            ],
          },
          sharing: {
            title: 'Vacations and Your Budget',
            message: '',
            url: 'https://apollosrock.newspring.cc/',
          },
        },
      ],
      name: 'Liked Content',
      navigation: jest.fn(),
      isLoading: false,
    };

    const tree = renderer.create(
      <Providers>
        <RecentlyLikedTileFeed {...props} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders a loading state', () => {
    const props = {
      content: [
        {
          id: 'UniversalContentItem:4148aeb9482181025e9cad42826e676a',
          title: 'Making Prayer a Spiritual Habit',
          coverImage: {
            sources: [
              {
                uri:
                  'https://apollosrock.newspring.cc/GetImage.ashx?guid=55be1fd5-d8eb-43a6-aa70-f8e27bb63d31',
              },
            ],
          },
          sharing: {
            title: 'Making Prayer a Spiritual Habit',
            message: '',
            url: 'https://apollosrock.newspring.cc/',
          },
        },
        {
          id: 'UniversalContentItem:9dbdc4e565dcaa3c9b17eb5ae42de96e',
          title: 'Vacations and Your Budget',
          coverImage: {
            sources: [
              {
                uri:
                  'https://apollosrock.newspring.cc/GetImage.ashx?guid=993efb47-47ec-4dec-a3d3-47bcacfbd58e',
              },
            ],
          },
          sharing: {
            title: 'Vacations and Your Budget',
            message: '',
            url: 'https://apollosrock.newspring.cc/',
          },
        },
      ],
      name: 'Liked Content',
      navigation: jest.fn(),
      isLoading: true,
    };

    const tree = renderer.create(
      <Providers>
        <RecentlyLikedTileFeed {...props} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
