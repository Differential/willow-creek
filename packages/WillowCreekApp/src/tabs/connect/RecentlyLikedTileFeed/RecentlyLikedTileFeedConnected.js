import React from 'react';
import { Query } from 'react-apollo';
import PropTypes from 'prop-types';

import getLikedContent from '../getLikedContent';
import RecentlyLikedTileFeed from './RecentlyLikedTileFeed';

const RecentlyLikedTileFeedConnected = ({ navigation }) => (
  <Query query={getLikedContent} fetchPolicy="cache-and-network">
    {({ loading, data: { getAllLikedContent = [] } = {} }) => {
      if (!getAllLikedContent.length) return null;
      return (
        <RecentlyLikedTileFeed
          id={'liked'}
          name={'Recently Liked'}
          content={getAllLikedContent}
          isLoading={loading}
          navigation={navigation}
          loadingStateObject={{
            title: 'Recently Liked',
            isLoading: true,
          }}
        />
      );
    }}
  </Query>
);

RecentlyLikedTileFeedConnected.propTypes = {
  navigation: PropTypes.shape({
    getParam: PropTypes.func,
    navigate: PropTypes.func,
  }),
};

export default RecentlyLikedTileFeedConnected;
