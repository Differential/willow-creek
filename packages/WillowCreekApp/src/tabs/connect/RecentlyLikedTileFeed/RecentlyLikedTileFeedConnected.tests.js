import React from 'react';

import Providers from '../../../Providers';

import { renderWithApolloData } from '../../../utils/testUtils';

import GET_LIKED_CONTENT from '../getLikedContent';
import RecentlyLikedTileFeedConnected from './RecentlyLikedTileFeedConnected';

describe('RecentlyLikedTileFeedConnected', () => {
  it('renders a RecentlyLikedTileFeedConnected', async () => {
    const mock = {
      request: {
        query: GET_LIKED_CONTENT,
        variables: { first: 3 },
      },
      result: {
        data: {
          likedContent: {
            __typename: 'ContentItemsConnection',
            pageInfo: {
              __typename: 'PaginationInfo',
              endCursor: '123',
            },
            edges: [
              {
                __typename: 'ContentItemsConnectionEdge',
                node: {
                  __typename: 'UniversalContentItem',
                  id: 'UniversalContentItem:d57994350b9d213866b24dea3a97433d',
                  coverImage: null,
                  parentChannel: {
                    id: 'ContentChannel:4f68015ba18662a7409d1219a4ce013e',
                    name: 'Editorial',
                    iconName: 'text',
                    __typename: 'ContentChannel',
                  },
                  title: 'Mea Animal Aperiam Ornatus Eu',
                  hyphenatedTitle: 'Mea Animal Aperiam Ornatus Eu',
                  summary: 'Bla bla bla',
                  theme: null,
                  isLiked: false,
                  likedCount: 0,
                  videos: [],
                  audios: [],
                },
              },
              {
                __typename: 'ContentItemsConnectionEdge',
                node: {
                  __typename: 'UniversalContentItem',
                  id: 'UniversalContentItem:b36e55d803443431e96bb4b5068147ec',
                  coverImage: null,
                  parentChannel: {
                    id: 'ContentChannel:4f68015ba18662a7409d1219a4ce013e',
                    name: 'Editorial',
                    iconName: 'text',
                    __typename: 'ContentChannel',
                  },
                  title: 'Probo Senserit Id Mea, Ut Sed Malis Postea,',
                  hyphenatedTitle:
                    'Probo Senserit Id Mea, Ut Sed Malis Postea,',
                  summary: 'Bla bla bla',
                  theme: null,
                  isLiked: false,
                  likedCount: 0,
                  videos: [],
                  audios: [],
                },
              },
            ],
          },
        },
      },
    };
    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    const tree = await renderWithApolloData(
      <Providers mocks={[mock]}>
        <RecentlyLikedTileFeedConnected navigation={navigation} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders nothing if no liked content', async () => {
    const mock = {
      request: {
        query: GET_LIKED_CONTENT,
      },
      result: {
        data: {
          likedContent: {
            __typename: 'ContentItemsConnection',
            pageInfo: {
              __typename: 'PaginationInfo',
              endCursor: null,
            },
            edges: [],
          },
        },
      },
    };
    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    const tree = await renderWithApolloData(
      <Providers mocks={[mock]}>
        <RecentlyLikedTileFeedConnected navigation={navigation} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
