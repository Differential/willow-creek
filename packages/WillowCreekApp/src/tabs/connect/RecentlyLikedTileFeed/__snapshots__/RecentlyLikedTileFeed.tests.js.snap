// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentlyLikedTileFeed renders a RecentlyLikedTileFeed 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      horizontal={false}
      style={
        Object {
          "paddingHorizontal": 0,
          "paddingVertical": 16,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Bold",
                "fontSize": 16,
                "lineHeight": 23.04,
              }
            }
          >
            Liked Content
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row-reverse",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "System",
                "fontSize": 14,
                "fontStyle": null,
                "fontWeight": null,
                "lineHeight": 20.16,
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Object {
                  "color": "#17B582",
                }
              }
            >
              View All
            </Text>
          </Text>
        </View>
      </View>
      <RCTScrollView
        data={
          Array [
            Object {
              "__typename": "UniversalContentItem",
              "coverImage": Object {
                "sources": Array [
                  Object {
                    "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=55be1fd5-d8eb-43a6-aa70-f8e27bb63d31",
                  },
                ],
              },
              "id": "UniversalContentItem:4148aeb9482181025e9cad42826e676a",
              "sharing": Object {
                "message": "",
                "title": "Making Prayer a Spiritual Habit",
                "url": "https://apollosrock.newspring.cc/",
              },
              "title": "Making Prayer a Spiritual Habit",
            },
            Object {
              "__typename": "UniversalContentItem",
              "coverImage": Object {
                "sources": Array [
                  Object {
                    "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=993efb47-47ec-4dec-a3d3-47bcacfbd58e",
                  },
                ],
              },
              "id": "UniversalContentItem:9dbdc4e565dcaa3c9b17eb5ae42de96e",
              "sharing": Object {
                "message": "",
                "title": "Vacations and Your Budget",
                "url": "https://apollosrock.newspring.cc/",
              },
              "title": "Vacations and Your Budget",
            },
          ]
        }
        decelerationRate="fast"
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={5}
        initialScrollIndex={0}
        keyExtractor={[Function]}
        loadingStateObject={
          Object {
            "coverImage": Array [],
            "id": "fake_id",
            "title": "",
          }
        }
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        refreshing={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        snapToAlignment="start"
        snapToInterval={592}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "height": 600,
            "paddingBottom": 8,
          }
        }
        tileHeight={600}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "padding": 8,
                  "paddingLeft": 16,
                  "paddingRight": 0,
                }
              }
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "aspectRatio": 1,
                      "borderRadius": 16,
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#A5A5A5",
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      animate={null}
                      customAnimate={null}
                      onReady={false}
                      style={
                        Object {
                          "aspectRatio": 1,
                          "backgroundColor": "#A5A5A5",
                          "width": "100%",
                        }
                      }
                    >
                      <Image
                        onLoad={[Function]}
                        source={
                          Array [
                            Object {
                              "cache": "force-cache",
                              "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=55be1fd5-d8eb-43a6-aa70-f8e27bb63d31",
                            },
                          ]
                        }
                        style={
                          Object {
                            "aspectRatio": 1,
                            "backgroundColor": "#A5A5A5",
                            "opacity": 0,
                            "resizeMode": "cover",
                            "width": "100%",
                          }
                        }
                      />
                    </View>
                    <BVLinearGradient
                      colors={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      endPoint={
                        Object {
                          "x": 0,
                          "y": 1,
                        }
                      }
                      locations={
                        Array [
                          0.3,
                          1,
                        ]
                      }
                      startPoint={
                        Object {
                          "x": 0,
                          "y": 0,
                        }
                      }
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                  </View>
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 16,
                        "color": "#F8F7F4",
                        "fontFamily": "InterUI-Bold",
                        "fontSize": 16,
                        "left": 16,
                        "lineHeight": 23.04,
                        "position": "absolute",
                        "right": 16,
                      }
                    }
                  >
                    Making Prayer a Spiritual Habit
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "padding": 8,
                  "paddingLeft": 16,
                  "paddingRight": 0,
                }
              }
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "aspectRatio": 1,
                      "borderRadius": 16,
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#A5A5A5",
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      animate={null}
                      customAnimate={null}
                      onReady={false}
                      style={
                        Object {
                          "aspectRatio": 1,
                          "backgroundColor": "#A5A5A5",
                          "width": "100%",
                        }
                      }
                    >
                      <Image
                        onLoad={[Function]}
                        source={
                          Array [
                            Object {
                              "cache": "force-cache",
                              "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=993efb47-47ec-4dec-a3d3-47bcacfbd58e",
                            },
                          ]
                        }
                        style={
                          Object {
                            "aspectRatio": 1,
                            "backgroundColor": "#A5A5A5",
                            "opacity": 0,
                            "resizeMode": "cover",
                            "width": "100%",
                          }
                        }
                      />
                    </View>
                    <BVLinearGradient
                      colors={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      endPoint={
                        Object {
                          "x": 0,
                          "y": 1,
                        }
                      }
                      locations={
                        Array [
                          0.3,
                          1,
                        ]
                      }
                      startPoint={
                        Object {
                          "x": 0,
                          "y": 0,
                        }
                      }
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                  </View>
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 16,
                        "color": "#F8F7F4",
                        "fontFamily": "InterUI-Bold",
                        "fontSize": 16,
                        "left": 16,
                        "lineHeight": 23.04,
                        "position": "absolute",
                        "right": 16,
                      }
                    }
                  >
                    Vacations and Your Budget
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`RecentlyLikedTileFeed renders a loading state 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      horizontal={false}
      style={
        Object {
          "paddingHorizontal": 0,
          "paddingVertical": 16,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#A5A5A5",
                "borderRadius": 16,
                "height": 16,
                "marginVertical": 3.5199999999999996,
                "width": "80%",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flexDirection": "row-reverse",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#A5A5A5",
                "borderRadius": 16,
                "height": 14,
                "marginVertical": 3.08,
                "width": "40%",
              }
            }
          />
        </View>
      </View>
      <RCTScrollView
        data={
          Array [
            Object {
              "coverImage": Object {
                "sources": Array [
                  Object {
                    "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=55be1fd5-d8eb-43a6-aa70-f8e27bb63d31",
                  },
                ],
              },
              "id": "UniversalContentItem:4148aeb9482181025e9cad42826e676a",
              "sharing": Object {
                "message": "",
                "title": "Making Prayer a Spiritual Habit",
                "url": "https://apollosrock.newspring.cc/",
              },
              "title": "Making Prayer a Spiritual Habit",
            },
            Object {
              "coverImage": Object {
                "sources": Array [
                  Object {
                    "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=993efb47-47ec-4dec-a3d3-47bcacfbd58e",
                  },
                ],
              },
              "id": "UniversalContentItem:9dbdc4e565dcaa3c9b17eb5ae42de96e",
              "sharing": Object {
                "message": "",
                "title": "Vacations and Your Budget",
                "url": "https://apollosrock.newspring.cc/",
              },
              "title": "Vacations and Your Budget",
            },
          ]
        }
        decelerationRate="fast"
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={5}
        initialScrollIndex={0}
        keyExtractor={[Function]}
        loadingStateObject={
          Object {
            "coverImage": Array [],
            "id": "fake_id",
            "title": "",
          }
        }
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        refreshing={true}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        snapToAlignment="start"
        snapToInterval={592}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "height": 600,
            "paddingBottom": 8,
          }
        }
        tileHeight={600}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "padding": 8,
                  "paddingLeft": 16,
                  "paddingRight": 0,
                }
              }
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "aspectRatio": 1,
                      "borderRadius": 16,
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#A5A5A5",
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      animate={null}
                      customAnimate={null}
                      onReady={false}
                      style={
                        Object {
                          "aspectRatio": 1,
                          "backgroundColor": "#A5A5A5",
                          "width": "100%",
                        }
                      }
                    >
                      <Image
                        onLoad={[Function]}
                        source={
                          Array [
                            Object {
                              "cache": "force-cache",
                              "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=55be1fd5-d8eb-43a6-aa70-f8e27bb63d31",
                            },
                          ]
                        }
                        style={
                          Object {
                            "aspectRatio": 1,
                            "backgroundColor": "#A5A5A5",
                            "opacity": 0,
                            "resizeMode": "cover",
                            "width": "100%",
                          }
                        }
                      />
                    </View>
                    <BVLinearGradient
                      colors={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      endPoint={
                        Object {
                          "x": 0,
                          "y": 1,
                        }
                      }
                      locations={
                        Array [
                          0.3,
                          1,
                        ]
                      }
                      startPoint={
                        Object {
                          "x": 0,
                          "y": 0,
                        }
                      }
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "alignSelf": "stretch",
                        "backgroundColor": "#A5A5A5",
                        "borderRadius": 16,
                        "height": 16,
                        "marginVertical": 3.5199999999999996,
                        "width": "80%",
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "padding": 8,
                  "paddingLeft": 16,
                  "paddingRight": 0,
                }
              }
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "aspectRatio": 1,
                      "borderRadius": 16,
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#A5A5A5",
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      animate={null}
                      customAnimate={null}
                      onReady={false}
                      style={
                        Object {
                          "aspectRatio": 1,
                          "backgroundColor": "#A5A5A5",
                          "width": "100%",
                        }
                      }
                    >
                      <Image
                        onLoad={[Function]}
                        source={
                          Array [
                            Object {
                              "cache": "force-cache",
                              "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=993efb47-47ec-4dec-a3d3-47bcacfbd58e",
                            },
                          ]
                        }
                        style={
                          Object {
                            "aspectRatio": 1,
                            "backgroundColor": "#A5A5A5",
                            "opacity": 0,
                            "resizeMode": "cover",
                            "width": "100%",
                          }
                        }
                      />
                    </View>
                    <BVLinearGradient
                      colors={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      endPoint={
                        Object {
                          "x": 0,
                          "y": 1,
                        }
                      }
                      locations={
                        Array [
                          0.3,
                          1,
                        ]
                      }
                      startPoint={
                        Object {
                          "x": 0,
                          "y": 0,
                        }
                      }
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "alignSelf": "stretch",
                        "backgroundColor": "#A5A5A5",
                        "borderRadius": 16,
                        "height": 16,
                        "marginVertical": 3.5199999999999996,
                        "width": "80%",
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTSafeAreaView>
`;
